### ISSUE TITLE
Joystick naming

### ISSUE THREAD

**Issue by inferrna** [aardappel/lobster#277]
I tried to test my **F510** with lobster game engine and found that I can't detect one of buttons with simple enumeration from 0 to 20. It suddenly became that on my system (ClearLinux) `'0' + 10;` is not `'10'` but `':'`.
So it looks like more convenient to use to_string()

```diff
diff --git a/dev/src/sdlsystem.cpp b/dev/src/sdlsystem.cpp
index b4c1b32..222385d 100644
--- a/dev/src/sdlsystem.cpp
+++ b/dev/src/sdlsystem.cpp
@@ -712,8 +712,7 @@ bool SDLFrame() {
 
             case SDL_JOYBUTTONDOWN:
             case SDL_JOYBUTTONUP: {
-                string name = "joy";
-                name += '0' + (char)event.jbutton.button;
+                string name = "joy" + to_string(event.jbutton.button);
                 updatebutton(name, event.jbutton.state == SDL_PRESSED, 0, false);
                 break;
             }
```

**Comment by aardappel** [comment]
Yup, your code is better.. I guess that was originally written with an assumption there's not more than 10 buttons ever :)

Wanna add it in a PR?

Also, we just added a better joystick API.. instead of buttons starting with `joy`, use the one starting with `controller_`.

**Comment by inferrna** [comment]
The PR https://github.com/aardappel/lobster/pull/279

>I guess that was originally written with an assumption there's not more than 10 buttons ever :)

My F510 has an additional button which summons steam window (I suppose it's emulating the x-box "guide" button). Actually, I don't need it and would be ok with ":", but ":" was on the right thumbstick which is my favorite one ))

>instead of buttons starting with joy, use the one starting with controller_

I see the line 
```c++
auto sdl_name = SDL_GameControllerGetStringForButton((SDL_GameControllerButton)event.cbutton.button)
```
But how can I see the name of actually pressed button? For reassignment, for example. I expect something like `gl.last_pressed_button()` returning string.

**Comment by aardappel** [comment]
That should simply be added to the documentation somewhere.. the names are here for now: https://github.com/aardappel/lobster/blob/edb79d1ff198ea3fe9d3a7dc55deb82fce96cabd/dev/src/sdlsystem.cpp#L58-L63
